libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookHelper.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder.Adapters.Facebook.FacebookEvents;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Adapters.Facebook
+{
+    public static class FacebookHelper
+    {
+        /// <summary>
+        /// Converts an Activity object to a Facebook messenger outbound message ready for the API.
+        /// </summary>
+        /// <param name="activity">The activity to be converted to Facebook message.</param>
+        /// <returns>The resulting message.</returns>
+        public static FacebookMessage ActivityToFacebook(Activity activity)
+        {
+            if (activity == null)
+            {
+                return null;
+            }
+
+            var facebookMessage = new FacebookMessage(activity.Conversation.Id, new Message(), "RESPONSE");
+
+            facebookMessage.Message.Text = activity.Text;
+
+            // map these fields to their appropriate place
+            if (activity.ChannelData != null)
+            {
+                facebookMessage = activity.GetChannelData<FacebookMessage>();
+
+                // make sure the quick reply has a type
+                if (activity.GetChannelData<FacebookMessage>().Message.QuickReplies != null)
+                {
+                    facebookMessage.Message.QuickReplies = activity.GetChannelData<FacebookMessage>().Message.QuickReplies; // TODO: Add the content_type depending of what shape quick_replies has
+                }
+            }
+
+            if (activity.Attachments != null && activity.Attachments.Count > 0)
+            {
+                if (activity.Attachments.Count > 1)
+                {
+                    throw new Exception("Facebook message can only contain one attachment");
```

Comment:
Good point. Perhaps we shouldn't make the program throw an exception, but maybe we should have a way to inform the user of the one attachment limit. Right now we are not using any kind of logging.
Anyway, unless @gabog and @cleemullins eventually decide otherwise, we'll go with leaving the exception in for now, since this is the same way the attachment limit is handled in the Webex adapter.

Reply:
I'll pick this up with @cleemullins / @gabog - we are in a position where we can amend both webex and facebook. Leave it with me to pick this up.

Reply:
I discussed this with Chris / Gabo.  For now please stop throwing an exception and simply ignore any additional attachments. We might introduce some additional checks / logging in the next release (this is going out as a preview package with 4.6 so we still have the opportunity to improve in the coming weeks).